src/components/containers/visibility.tsx

```diff
@@ -125,38 +132,39 @@ export const Mobile = ({
     )
 }
 
-export const EU = ({ children }: ResponsiveContainerProps) => {
-    const { is_eu } = getCountryRule()
-
-    return is_eu ? <>{children}</> : null
-}
-
-export const NonEU = ({ children }: ResponsiveContainerProps) => {
-    const { is_non_eu } = getCountryRule()
+const CountryBasedContent = ({ country_rule, children }: CountryBasedContentProps) => {
+    const rules = getCountryRule()
+    const [is_loaded, setLoaded] = useState(false)
 
-    return is_non_eu ? <>{children}</> : null
-}
+    useLayoutEffect(() => {
+        setLoaded(true)
+    }, [rules])
 
-export const UK = ({ children }: ResponsiveContainerProps) => {
-    const { is_uk } = getCountryRule()
+    const condition = rules[country_rule]
 
-    return is_uk ? <>{children}</> : null
+    return is_loaded && condition ? <>{children}</> : null
 }
 
-export const NonUK = ({ children }: ResponsiveContainerProps) => {
-    const { is_non_uk } = getCountryRule()
+export const EU = ({ children }: ResponsiveContainerProps) => (
+    <CountryBasedContent country_rule="is_eu">{children}</CountryBasedContent>
+)
 
-    return is_non_uk ? <>{children}</> : null
-}
+export const NonEU = ({ children }: ResponsiveContainerProps) => (
+    <CountryBasedContent country_rule="is_non_eu">{children}</CountryBasedContent>
+)
 
-export const UKEU = ({ children }: ResponsiveContainerProps) => {
-    const { is_eu_uk } = getCountryRule()
+export const UK = ({ children }: ResponsiveContainerProps) => (
+    <CountryBasedContent country_rule="is_uk">{children}</CountryBasedContent>
+)
 
-    return is_eu_uk ? <>{children}</> : null
-}
+export const NonUK = ({ children }: ResponsiveContainerProps) => (
+    <CountryBasedContent country_rule="is_non_uk">{children}</CountryBasedContent>
+)
 
-export const ROW = ({ children }: ResponsiveContainerProps) => {
-    const { is_row } = getCountryRule()
+export const UKEU = ({ children }: ResponsiveContainerProps) => (
+    <CountryBasedContent country_rule="is_eu_uk">{children}</CountryBasedContent>
+)
```

Comment:
Only inconsistency I see is this bit. Are we able to change it globally you think?